src/ray/common/task/task_spec.h

```diff
@@ -61,6 +61,10 @@ inline bool operator==(const ray::rpc::SchedulingStrategy &lhs,
             rhs.placement_group_scheduling_strategy()
                 .placement_group_capture_child_tasks());
   }
+  case ray::rpc::SchedulingStrategy::kNodeLabelSchedulingStrategy: {
+    return (lhs.node_label_scheduling_strategy().SerializeAsString() ==
```

Comment:
Seems we cannot use SerializeAsString() since it's not stable per https://protobuf.dev/programming-guides/encoding/#order.

You can use `google::protobuf::util::MessageDifferencer::Equivalent` and we have examples in the codebase